src/sentry/replays/endpoints/organization_replay_viewed_by.py

```diff
@@ -0,0 +1,132 @@
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=ReplayExamples.GET_REPLAY_VIEWED_BY,
+    )
+    def get(self, request: Request, organization: Organization, replay_id: str) -> Response:
+        """
+        Returns list of serialized User objects who have viewed a given replay.
+        """
+        if not features.has("organizations:session-replay", organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            replay_id = str(uuid.UUID(replay_id))
+        except ValueError:
+            return Response(status=404)
+
+        try:
+            filter_params = self.get_filter_params(
+                request, organization, project_ids=ALL_ACCESS_PROJECTS
+            )
+        except NoProjects:
+            return Response(status=404)
+
+        # TODO: do we need this range filter?
+        if not filter_params["start"] or not filter_params["end"]:
+            return Response(status=404)
+
+        viewed_by_ids: list[int] = query_replay_viewed_by_ids(
+            project_id=filter_params["project_id"],
+            replay_id=replay_id,
+            start=filter_params["start"],
+            end=filter_params["end"],
+            organization=organization,
+        )
+        if not viewed_by_ids:
+            return Response(status=404)
+
+        response = generate_viewed_by_response(
+            replay_id=replay_id, viewed_by_ids=viewed_by_ids, actor=request.user
+        )
+        return Response({"data": response}, status=200)
+
+    @extend_schema(
+        operation_id="Signal the requesting user has viewed a replay",
+        parameters=[GlobalParams.ORG_SLUG, ReplayParams.REPLAY_ID],
+        responses={
+            200: RESPONSE_SUCCESS,
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=None,
+    )
+    def post(self, request: Request, organization: Organization, replay_id: str) -> Response:
+        """
+        Publishes a replay_viewed event for Snuba processing.
+        """
+        if not features.has("organizations:session-replay", organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            replay_id = str(uuid.UUID(replay_id))
+        except ValueError:
+            return Response(status=404)
+
+        try:
+            filter_params = self.get_filter_params(
+                request, organization, project_ids=ALL_ACCESS_PROJECTS
+            )
+        except NoProjects:
+            return Response(status=404)
+
+        # TODO: do we need this range filter?
+        if not filter_params["start"] or not filter_params["end"]:
+            return Response(status=404)
+
+        # TODO: are there more efficient ways to lookup or pass in the project_id, rather than a snuba query?
```

Comment:
yeah, why don't we just pass in the project_id? could even make this a project scoped endpoint

Reply:
So move both GET and POST to a project endpoint? Makes sense to me. I was following the blueprint for this draft